volatility/framework/plugins/windows/psscan.py

```diff
@@ -61,17 +69,88 @@ def scan_processes(cls,
+                Iterable[interfaces.objects.ObjectInterface]:
+        """ Returns a virtual process from a physical addressed one
+
+        Args:
+            context: The context to retrieve required elements (layers, symbol tables) from
+            layer_name: The name of the layer on which to operate
+            symbol_table: The name of the table containing the kernel symbols
+            proc: the process object with phisical address
 
+        Returns:
+            A process object on virtual address space
+
+        """
+
+        # We'll use the first thread to bounce back to the virtual process
+        kvo = context.layers[layer_name].config['kernel_virtual_offset']
+        ntkrnlmp = context.module(symbol_table, layer_name = layer_name, offset = kvo)
+
+        tleoffset = ntkrnlmp.get_type("_ETHREAD").relative_child_offset("ThreadListEntry")
+        # Start out with the member offset given to us from the profile
+        offsets = [tleoffset]
+
+        # If (and only if) we're dealing with 64-bit Windows 7 SP1
+        # then add the other commonly seen member offset to the list
+        kuser = info.Info.get_kuser_structure(context, layer_name, symbol_table)
+        nt_major_version = int(kuser.NtMajorVersion)
+        nt_minor_version = int(kuser.NtMinorVersion)
+        vers = info.Info.get_version_structure(context, layer_name, symbol_table)
+        build = vers.MinorVersion
+        bits = context.layers[layer_name].bits_per_register
+        version = (nt_major_version, nt_minor_version, build)
+        if version == (6, 1, 7601) and bits == 64:
+            offsets.append(tleoffset + 8)
+
+        # Now we can try to bounce back
+        for ofs in offsets:
+            ethread = ntkrnlmp.object(object_type = "_ETHREAD",
+                                      offset = proc.ThreadListHead.Flink - ofs,
+                                      absolute = True)
+
+            # Ask for the thread's process to get an _EPROCESS with a virtual address space
+            virtual_process = ethread.owning_process()
+            # Sanity check the bounce.
+            # This compares the original offset with the new one (translated from virtual space)
+            (_, _, ph_offset, _, _) = list(context.layers[layer_name].mapping(offset = virtual_process.vol.offset,
+                                                                              length = 0))[0]
+            if virtual_process and \
+               proc.vol.offset == ph_offset:
+                return virtual_process
+
+    def _generator(self):
+        pe_table_name = intermed.IntermediateSymbolTable.create(self.context,
+                                                                self.config_path,
+                                                                "windows",
+                                                                "pe",
+                                                                class_types = extensions.pe.class_types)
```

Comment:
I've had some issues with pe not being automatically imported when extensions is.  I've moved to importing `pe` directly, but if this works for you, then I'm happy for you to leave it as is...

Reply:
ok, thanks for the suggestion. I commited the changes